techniques/android/MASTG-TECH-0109.md

```diff
@@ -0,0 +1,51 @@
+--- 
+title: Intercepting Flutter HTTP Traffic
+platform: android 
+---
+
+Flutter is an open-source UI software development kit (SDK) created by Google. It is used for building natively compiled applications for mobile, web, and desktop from a single codebase. Flutter uses Dart, which is not proxy-aware and uses its own certificate store. The application doesn't take proxy configuration from the system and send the data directly to the server. Due to this, it is not possible to intercept the request using the BurpSuite or any MITM tools.
+
+
+
+**How does re-flutter method differs from other techniques ?**
```

Comment:
@sk3l10x1ng thanks for all the updates, I've fixed a couple of typos for now.


Also thanks for checking those blog posts. I think the only think that remains open is the consideration of the 3rd alternative technique. If I'm not misunderstanding anything they are 2 alternative approaches, 
- (re-flutter) requires patching 
- (nviso script) doesn't require patching, it uses Frida

Using these apps:
- https://github.com/NVISOsecurity/disable-flutter-tls-verification/blob/main/test_app/pinning.apk
- https://github.com/NVISOsecurity/disable-flutter-tls-verification/blob/main/test_app/pinning.ipa

Could you please validate that your technique works with them and also using this alternative technique via Frida, it seems easy to run using codeshare (as indicated [here](https://github.com/NVISOsecurity/disable-flutter-tls-verification)):

```
frida -U --codeshare TheDauntless/disable-flutter-tls-v1 -f YOUR_BINARY
```


If all of this works we can confirm that both techniques are equivalent and could be summarized as:

## re-flutter
1. setup interception proxy
2. patch app using re-flutter
3. inspect intercepted traffic

## nviso
1. setup interception proxy
2. run frida script
3. inspect intercepted traffic

Reply:
@TheDauntless Would you mind chiming in to confirm my hypothesis (or not)?